### PR TITLE
Fix Request::call to accept a URL parameter and add Content-Type header

### LINKED ISSUE #4: How to send a request?
In Request::call, the request is always posted to the "/" URL.

How does one make an XMLRPC request to a remote server? From reading the source for hyper, I didn't see any straightforward way to bind a hyper::client::Client to a particular base URL... it looks to me like the Client expects absolute URLs to be passed to get(), post(), etc., not relative URLs.

I am thinking that either Request::call needs to be modified to accept a target URL as a parameter, or if the method is actually correct as-is, some example code is needed for how to use it.

Also, I think that the Content-Type header should get set in the post():

`.header(ContentType(Mime(TopLevel::Text, SubLevel::Xml, vec![(Attr::Charset, Value::Utf8)])))`

### PR DESCRIPTION
**Pull request by bwbuchanan**
Fixes #4

### CODE REVIEW

File: src/request.rs [hunk 1]
```diff
@@ -37,18 +37,20 @@ impl<'a> Request<'a> {
...
     /// Returns a `RequestResult` indicating whether the request was sent and processed successfully
     /// (according to the rules of XML-RPC).
-    pub fn call(self, client: Client) -> RequestResult {
+    pub fn call(self, client: Client, url: &str) -> RequestResult {
         // First, build the body XML
         let mut body = Vec::new();
         try!(self.write_as_xml(&mut body));
 
+        use hyper::header::ContentType;
+        use hyper::mime::{Mime, TopLevel, SubLevel, Attr, Value};
```
Comment (jonas-schievink) [inline src/request.rs]: You can move these to the top of the file. I like having all imports at the top in most cases.

**Review by jonas-schievink — CHANGES_REQUESTED** [review]
That was fast! Thanks!

File: src/request.rs (hunk 1 above)
Comment (bwbuchanan) [reply src/request.rs]: hyper::mime::Value conflicts with xmlrpc::Value, and all of those symbols are used only within that method... which is why I stuck the "use" there.

Maybe promote the ContentType header to the top-of-file use block, but leave the hyper::mime use inside the method?

Or use the full path for hyper::mime::Value?

Which do you prefer?

Comment (jonas-schievink) [reply src/request.rs]: Hmm, okay.

> Maybe promote the ContentType header to the top-of-file use block, but leave the hyper::mime use inside the method?

Let's do this instead.

Comment (bwbuchanan) [reply src/request.rs]: OK! Already pushed revised commits.

**Review by jonas-schievink — APPROVED** [review]

**Pull request merged**